android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentHelper.java

```diff
@@ -22,164 +32,487 @@ public class ReceiveSharingIntentHelper {
           file.putString("mimeType", contentResolver.getType(uri));
-          Cursor queryResult = contentResolver.query(uri, null, null, null, null);
+          Cursor queryResult = contentResolver.query(
+            uri,
+            null,
+            null,
+            null,
+            null
+          );
           queryResult.moveToFirst();
-          file.putString("fileName", queryResult.getString(queryResult.getColumnIndex(OpenableColumns.DISPLAY_NAME)));
+          file.putString(
+            "fileName",
+            queryResult.getString(
+              queryResult.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+            )
+          );
           file.putString("filePath", filePath);
-          file.putString("contentUri",uri.toString());
-          file.putString("text",null);
+          file.putString("contentUri", uri.toString());
+          file.putString("text", null);
           file.putString("weblink", null);
           file.putString("subject", subject);
-          files.putMap(Integer.toString(index),file);
+
+          WritableMap exif = getExif(uri.toString());
+          files.putMap("exif", exif);
+
+          files.putMap(Integer.toString(index), file);
+
           index++;
         }
       }
     }
-    return  files;
+    return files;
+  }
+
+  public WritableMap getExif(String uri) {
+    try {
+      ExifInterface exif = createExifInterface(uri);
+
+      WritableMap exifMap = new WritableNativeMap();
+
+      for (String attribute : EXIF_ATTRIBUTES) {
+        String value = exif.getAttribute(attribute);
+
+        // Rename props for convenience
+        if (attribute == ExifInterface.TAG_IMAGE_LENGTH) {
+          exifMap.putString("height", value);
+        } else if (attribute == ExifInterface.TAG_IMAGE_WIDTH) {
+          exifMap.putString("width", value);
+        } else {
+          exifMap.putString(attribute, value);
+        }
+      }
+
+      exifMap.putString("originalUri", uri);
+
+      return exifMap;
+    } catch (Exception e) {
+      WritableMap exifMap = new WritableNativeMap();
+      return exifMap;
+    }
+  }
+
+  public static Uri compatUriFromFile(final Context context, final File file) {
+    Uri result = null;
+    if (Build.VERSION.SDK_INT < 21) {
+      result = Uri.fromFile(file);
+    } else {
+      final String packageName = context
+        .getApplicationContext()
+        .getPackageName();
+      final String authority = new StringBuilder(packageName)
+        .append(".provider")
+        .toString();
+      try {
+        result = FileProvider.getUriForFile(context, authority, file);
+      } catch (IllegalArgumentException e) {
+        e.printStackTrace();
+      }
+    }
+    return result;
+  }
+
+  @SuppressLint("NewApi")
+  public static String getRealPathFromURI(
+    final Context context,
+    final Uri uri
+  ) {
+    final boolean isKitKat =
+      Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT;
```

Comment:
In our app minSdkVersion is 24 - Nougat

I think we can safely drop those "ancient" logics. Wdyt?

Reply:
Agree! Removed in both places